robusta_krr/core/integrations/prometheus/metrics/memory_metric.py

```diff
@@ -0,0 +1,60 @@
+import datetime
+from typing import Any, Optional
+
+from robusta_krr.core.models.allocations import ResourceType
+
+from .base_metric import BaseMetricLoader, bind_metric
+
+PrometheusSeries = Any
+
+
+@bind_metric(ResourceType.Memory)
+class MemoryMetricLoader(BaseMetricLoader):
+    @staticmethod
+    def get_target_name(series: PrometheusSeries) -> Optional[str]:
+        for label in ["container", "pod", "node"]:
+            if label in series["metric"]:
+                return series["metric"][label]
+        return None
+
+    @staticmethod
+    def filter_prom_jobs_results(
+        series_list_result: list[PrometheusSeries],
+    ) -> list[PrometheusSeries]:
+        """
+        Because there might be multiple metrics with the same name, we need to filter them out.
+
```

Comment:
I would comment why we do it here and not for CPU
Cpu usually only comes from node-exporter job